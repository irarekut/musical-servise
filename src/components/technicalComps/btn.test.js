/* eslint-disable no-restricted-globals */
/* eslint-disable no-undef */
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import BtnLogin from './btn'

describe('<BtnLogin />', () => {
  describe('Snapshots tests', () => {
    it('should render successfully and match snapshot', () => {
      const { container } = render(<BtnLogin>Войти</BtnLogin>)

      expect(container).toMatchSnapshot()
    })
  })
  describe('Attributes tests', () => {
    it('should set type="submit" by default', () => {
      render(<BtnLogin />)

      expect(screen.getByRole('button')).toHaveAttribute('type', 'submit')
    })
  })
  describe('Callbacks tests', () => {
    it("should call 'onClick' prop", () => {
      const cb = jest.fn()

      render(<BtnLogin onClick={cb} />)

      fireEvent.click(screen.getByRole('button'))

      expect(cb).toBeCalledTimes(1)
    })
  })
})
