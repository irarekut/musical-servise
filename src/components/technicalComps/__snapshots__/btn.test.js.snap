// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BtnLogin /> Snapshots tests should render successfully and match snapshot 1`] = `
<div>
  <button
    class="sc-bcXHqe hITTHK"
    type="submit"
  />
</div>
`;
